### PR TITLE
Ball tensioner

### PR DESCRIPTION
**Pull request by Team2890**
Added support for the ball tensioner on the intake manipulator

### CODE REVIEW

**Comment by Team2890** [comment]
Hey
uhm
Check out my code
What should we add?

**Comment by Valance23322** [comment]
The README template should not be included in this repo

File: src/main/java/frc/robot/RobotMap.java [hunk 1]
```diff
@@ -131,6 +158,8 @@ public static void init()
 
     hatchHolder = new Servo(HATCH_HOLDER_PORT);
 
+    ballIntakeStop = new DigitalInput(BALL_INTAKE_STOP_PORT);
```
Comment (Valance23322) [inline src/main/java/frc/robot/RobotMap.java]: What is ballIntakeStop?  I'm assuming that it prevents the ball from shooting out the intake mechanism when we're picking up but what hardware component does this correspond to?

File: src/main/java/frc/robot/subsystems/ElevatorSubsystem.java [hunk 2]
```diff
@@ -25,24 +25,29 @@ public void initDefaultCommand()
...
+   * Moves the elevator up/down at the given speed. Positive for up, Negative for down
+   * @param speed
+   */
+  public void moveElevator(double speed)
   {
-    elevatorUp(-speed);
+    RobotMap.elevatorSparkMax.set(speed * RobotMap.ELEVATOR_SPEED_MODIFIER);
   }
 
   public double getEncoderPosition()
```
Comment (Valance23322) [inline src/main/java/frc/robot/subsystems/ElevatorSubsystem.java]: This is unrelated to this commit/pull request, but you probably want to add a javadoc explaining how to interpret the output of getEncoderPosition()

File: src/main/java/frc/robot/subsystems/ManipulatorSubsystem.java [hunk 3]
```diff
@@ -42,11 +42,23 @@ public void releaseHatch()
...
+    //Spins the motors when the button is pressed. THIS IS ONLY RUN ONCE UNTIL THE BUTTON IS RELEASED AND PRESSED AGAIN
+    if (RobotMap.assistantDriverController.getXButtonPressed())
+    {
+      spinIntake(RobotMap.MANIPULATOR_FULL_SPEED * RobotMap.MANIPULATOR_SPEED_MODIFER); //Spin the intake
+      RobotMap.ballInIntake = !RobotMap.ballInIntake;
+    }
+    //Only stop the motors when the button is released
+    else if (RobotMap.assistantDriverController.getXButtonReleased())
+      spinIntake(RobotMap.MANIPULATOR_STOP_SPEED); //Stop the intake from spinning
+    else if (!RobotMap.ballIntakeStop.get() && RobotMap.ballInIntake)
```
Comment (Valance23322) [inline src/main/java/frc/robot/subsystems/ManipulatorSubsystem.java]: When would ballIntakeStop.get() return false?

File: src/main/java/frc/robot/RobotMap.java (hunk 1 above)
Comment (Podcakero) [reply src/main/java/frc/robot/RobotMap.java]: its a micro switch that triggers when a ball is inside

File: src/main/java/frc/robot/subsystems/ManipulatorSubsystem.java (hunk 3 above)
Comment (Podcakero) [reply src/main/java/frc/robot/subsystems/ManipulatorSubsystem.java]: it depends on how the micro switch is wired. if its wired to when depressed equals false, then we dont need the !. if its the other way around we do

File: src/main/java/frc/robot/subsystems/ElevatorSubsystem.java (hunk 2 above)
Comment (Podcakero) [reply src/main/java/frc/robot/subsystems/ElevatorSubsystem.java]: i told alex to work on documentation. not sure how much we got done

File: src/main/java/frc/robot/subsystems/ManipulatorSubsystem.java (hunk 3 above)
Comment (Podcakero) [reply src/main/java/frc/robot/subsystems/ManipulatorSubsystem.java]: i will document this method more to show the logic and how it SHOULD work tomorrow

File: src/main/java/frc/robot/RobotMap.java (hunk 1 above)
Comment (Podcakero) [reply src/main/java/frc/robot/RobotMap.java]: should i change the name to include switch?

Comment (Team2890) [reply src/main/java/frc/robot/RobotMap.java]: Name has been changed to include switch

**Comment by Podcakero** [comment]
All the stuff i want done is done
And we're currently working on the elevator

**Comment by Podcakero** [comment]
Forget to merge

**Comment by Podcakero** [comment]
Objective finished
Even though idea changed

**Pull request merged**